nemo/collections/llm/gpt/data/pre_training.py

```diff
@@ -38,16 +38,30 @@ def __init__(
         index_mapping_dir: Optional[str] = None,
     ) -> None:
         super().__init__()
-        if not isinstance(paths, (list, tuple)):
+        if not isinstance(paths, (list, tuple, dict)):
             paths = [paths]
-        if weights is not None:
-            assert len(weights) == len(paths)
-            if len(weights) == 1:
-                # weights must be None if there is only one dataset
+
+        from megatron.core.datasets.utils import get_blend_from_list
+
+        build_kwargs = {}
+        if isinstance(paths, dict):
+            if split is not None:
+                warnings.warn(
+                    f"{split=} will be ignored since datasets are being created " f"from 3 separate distributions."
+                )
+            build_kwargs["blend_per_split"] = [
+                get_blend_from_list(paths["train"]),
+                get_blend_from_list(paths["validation"]),
+                get_blend_from_list(paths["test"]),
```

Comment:
@ashors1 does get_blend_from_list(paths["train"]) work even when you have multiple data files. For ex: {"train": /datafile1/, /datafile2/}. Also in this case if weights are ignored then is the dataset built with all samples from both /datafile1/ and /datafile2/ ?

Reply:
Yes, `get_blend_from_list(paths["train"])` works when you have multiple paths. You're also able to pass in weights by interleaving them with the paths. For example, the following would work: 
```
paths={
            "train": [25, PATH1, 75, PATH2],
            "validation": [PATH3, PATH4],
            "test": ['1', PATH5], 
        }
```

The only time the weights are not used is when `limit_val_batches <= 1.0`, in which case we want to return the full validation dataset. In this case, users are expected not to provide weights for the paths.